packages/admin-ui-hooks/src/use-intersection-observer.ts

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useRef, useState } from 'react'
+
+/**
+ * https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API
+ */
+export function useIntersectionObserver({
+  // Defaults to the browser viewport
+  root = null,
+  rootMargin = undefined,
+  threshold = 0,
+  callback,
+}: IntersectionObserverInit & {
+  callback: (entry: IntersectionObserverEntry) => void
+}) {
+  const [node, setNode] = useState<HTMLDivElement>()
+  const observer = useRef<IntersectionObserver>()
```

Comment:
Just to follow our code styleguide: [Avoid param destructuring](https://admin-ui.vercel.app/teams/developing/code-styleguide#avoid-param-destructuring) and [Avoid inline types](https://admin-ui.vercel.app/teams/developing/code-styleguide#avoid-inline-types)

```suggestion
export function useIntersectionObserver(props: UseIntersectionObserverProps) {
  const {
    // Defaults to the browser viewport
    root = null,
    rootMargin = undefined,
    threshold = 0,
    callback,
  } = props
  
  const [node, setNode] = useState<HTMLDivElement>()
  const observer = useRef<IntersectionObserver>()
```

Reply:
Done 😉